Add API test suites

Unweldify opened the pull request:
## Реализован автотест API БД сущностей.
### В проекте имплементированы:
### Сьют тестов `test_api`:
- Добавление сущности
- Удаление сущности
- Получение сущности
- Получение списка сущностей
- Замена данных сущности

### `Allure` для читабельности и описания шагов тестов.

<hr>

### Эта ветка будет слияться с веткой `main`, где собственно и будет находиться финальный проект автотестов.

Unweldify committed:
added api test suites

Unweldify committed:
Update README.md

Unweldify committed:
added requirements.txt

Unweldify committed:
Merge branch 'API_pytest' of https://github.com/Unweldify/API-SDET into API_pytest

StasovSt commented on Tests/test_api.py:
```diff
@@ -0,0 +1,81 @@
+
+
+@allure.title('Проверить функциональность добавления сущности.')
+@staticmethod
```
причем тут staticmethod?

StasovSt commented on Services/Entity/data_generator.py:
```diff
@@ -0,0 +1,22 @@
+
+class Generator:
+    """Класс генератор для создания данных"""
+    def generate_client_data():
```
Преобразовать функцию в staticmethod

StasovSt commented on Tests/test_api.py:
```diff
@@ -0,0 +1,81 @@
+@staticmethod
+def test_create_entity(entitymodel, entity_api, generate_data):
+    with allure.step('Генерируем cущество'):
+        entity = entitymodel(**generate_data())
```
вынести в фикстуру

Unweldify committed:
tweaked docstrings and steps
fully moved entity generation into a fixture
removed accidental @staticmethod occurences
added @staticmethod in data_generator

StasovSt commented on Services/Entity/data_generator.py:
```diff
@@ -0,0 +1,24 @@
+from random import randint
+from typing import Dict
+
+class Generator:
```
Этот класс относится к подготовке тестовых данных, а не к работе с api
Лучше перенести данный модуль в ./data/data_generator

StasovSt commented on Tests/test_api.py:
```diff
@@ -0,0 +1,65 @@
+
+@allure.title('Проверить функциональность добавления сущности.')
+def test_create_entity(entity_api, generated_entity):
+    id = int(entity_api.create_entity(generated_entity))
```
вынести в фикстуру

Unweldify replied:
Момент с int() был недосмотром с моей стороны, create_entity изначально должен был возвращать int
Думаю после исправления тут не стоит выносить в фикстуру. Но может я не допонял

StasovSt commented on Tests/test_api.py:
```diff
@@ -0,0 +1,65 @@
+    entities = entity_api.get_all_entities()
+
+    with allure.step('Посмотрим если был получен список сущностей.'):
+        assert entities != [], (
```
assert entities достаточно

StasovSt commented on Tests/test_api.py:
```diff
@@ -0,0 +1,65 @@
+
+    in_list = entity_api.get_entity(id)
+
+    with allure.step('Посмотрим если существо было получено'):
```
"Существо" - интересно конечно)) 
но лучше переименовать на "сущность"

StasovSt commented on Services/Entity/request_handler.py:
```diff
@@ -0,0 +1,37 @@
+load_dotenv()
+
+
+class Request:
```
Предлагаю данный класс и модуль перенести в ./api/base_api

StasovSt commented on Services/Entity/entity_request.py:
```diff
@@ -0,0 +1,43 @@
+request = Request()
+
+
+class EntityRequest:
```
Предлагаю данный класс и модуль перенести в ./api/entity_service
Сам класс переименовать EntityService

Unweldify committed:
restructured module locations
fixed create_entity not returning int id but json
fixed errors in step strings
